grails-app/services/org/pih/warehouse/data/LocationDataService.groovy

```diff
@@ -115,12 +130,20 @@ class LocationDataService {
         location.name = params.name
         location.active = Boolean.valueOf(params.active)
         location.locationNumber = params.locationNumber
-        location.locationType = params.locationType ? LocationType.findByNameLike(params.locationType + "%") : null
         location.locationGroup = params.locationGroup ? LocationGroup.findByName(params.locationGroup) : null
         location.parentLocation = params.parentLocation ? Location.findByNameOrLocationNumber(params.parentLocation, params.parentLocation) : null
         location.organization = params.organization ? Organization.findByCodeOrName(params.organization, params.organization) : null
         location.address = createOrUpdateAddress(params, location?.address?.id)
 
+        def locationType = params.locationType ? LocationType.findByNameLike(params.locationType + "%") : null
+        def currentLocationType = location.locationType
+
+        //Do not allow to change internal type location to non-internal
+        if (((currentLocationType?.isInternalLocation() || currentLocationType?.isZone()) && (locationType?.isInternalLocation() || locationType?.isZone())) || !(currentLocationType?.isInternalLocation() || currentLocationType?.isZone())) {
+            location.locationType = locationType
+            location.supportedActivities = currentLocationType && locationType == currentLocationType ? location.supportedActivities : location.supportedActivities?.clear()
```

Comment:
@MagdaToczek This makes me a little nervous (the assignment of the supported activities in the default case might mark the association as "dirty" and we'd like to avoid extra db activity if possible given Hibernate's penchant for lazy init errors and the like). I would just add some logic like this.
```
if (currentLocationType && locationType != currentLocationType) { 
    location.supportedActivities?.clear()
}
```
With that said, we can leave this for now. Just keep an eye on this during testing.